internal/topology/check/compatibility_test.go

```diff
@@ -170,12 +174,222 @@ func TestCheckReferencedObjectsAreStrictlyCompatible(t *testing.T) {
+				t.Errorf("LocalObjectTemplateAreCompatible() error = %v, wantErr %v", err.ToAggregate(), tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestLocalObjectTemplateIsValid(t *testing.T) {
+	namespace := metav1.NamespaceDefault
+	pathPrefix := field.NewPath("this", "is", "a", "prefix")
+
+	validTemplate := &clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "default",
+			Name:       "valid",
+			Kind:       "barTemplate",
+			APIVersion: "test.group.io/versionone",
+		},
+	}
+
+	nilTemplate := &clusterv1.LocalObjectTemplate{
+		Ref: nil,
+	}
+	emptyNameTemplate := &clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "default",
+			Name:       "",
+			Kind:       "barTemplate",
+			APIVersion: "test.group.io/versionone",
+		},
+	}
+	wrongNamespaceTemplate := &clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "wrongNamespace",
+			Name:       "foo",
+			Kind:       "barTemplate",
+			APIVersion: "test.group.io/versiontwo",
+		},
+	}
+	notTemplateKindTemplate := &clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "default",
+			Name:       "foo",
+			Kind:       "bar",
+			APIVersion: "test.group.io/versionone",
+		},
+	}
+	invalidAPIVersionTemplate := &clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "default",
+			Name:       "foo",
+			Kind:       "barTemplate",
+			APIVersion: "this/has/too/many/slashes",
+		},
+	}
+	emptyAPIVersionTemplate := &clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "default",
+			Name:       "foo",
+			Kind:       "barTemplate",
+			APIVersion: "",
+		},
+	}
+
+	tests := []struct {
+		template *clusterv1.LocalObjectTemplate
+		name     string
+		wantErr  bool
+	}{
+		{
+			name:     "No error with valid Template",
+			template: validTemplate,
+			wantErr:  false,
+		},
+
+		{
+			name:     "Invalid if ref is nil",
+			template: nilTemplate,
+			wantErr:  true,
+		},
+		{
+			name:     "Invalid if name is empty",
+			template: emptyNameTemplate,
+			wantErr:  true,
+		},
+		{
+			name:     "Invalid if namespace doesn't match",
+			template: wrongNamespaceTemplate,
+			wantErr:  true,
+		},
+		{
+			name:     "Invalid if Kind doesn't have Template suffix",
+			template: notTemplateKindTemplate,
+			wantErr:  true,
+		},
+		{
+			name:     "Invalid if apiVersion is not valid",
+			template: invalidAPIVersionTemplate,
+			wantErr:  true,
+		},
+		{
+			name:     "Empty apiVersion is not valid",
+			template: emptyAPIVersionTemplate,
+			wantErr:  true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			allErrs := LocalObjectTemplateIsValid(tt.template, namespace, pathPrefix).ToAggregate()
+			if tt.wantErr {
+				g.Expect(allErrs).To(HaveOccurred())
+				return
+			}
+			g.Expect(allErrs).ToNot(HaveOccurred())
+		})
+	}
+}
+
+// TODO: Add unit tests for new functions in compatibility package.
```

Comment:
Will this be done in this PR or follow-up? (should we create an issue for it if yes?)

Reply:
I'll add this to this PR - these tests were left pending the outcome of #5641